public/javascripts/main.js

```diff
@@ -35,12 +35,37 @@ $(function(defaultDateStart, defaultDateEnd){
 
 $(document).ready(function() {
     allSelected();
+
+    checkSelectedMeals();
 });
 
 $(".schedule").change(function(){
     allSelected();
 });
 
+function checkSelectedMeals() {
+    $(".schedule").change(function () {
+        if (this.checked) {
```

Comment:
You can simplify this block by just checking `$('input[day="dd-mm-yyyy"]:checked')` where `day` is a new attribute you add to the checkbox (in the twirl template) with the value of that meal. Then on the query you specify `$(this).getAttr('day')` to compare against.